Formula/swiftplantuml.rb

```diff
@@ -0,0 +1,18 @@
+class Swiftplantuml < Formula
+  desc "Generate UML class diagrams from Swift sources"
+  homepage "https://github.com/MarcoEidinger/SwiftPlantUML"
+  url "https://github.com/MarcoEidinger/SwiftPlantUML/archive/0.5.0.tar.gz"
+  sha256 "2ba045c719e11d84bb8951b7e352b34e7d459318505f8fc9b1987328de598dff"
+  license "MIT"
+
+  depends_on xcode: ["12.2", :build]
+  depends_on :macos
+
+  def install
+    system "make", "install", "prefix=#{prefix}"
+  end
+
+  test do
+    system "#{bin}/swiftplantuml", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@chenrui333 would the following be sufficient? 

```ruby
  test do
    (testpath/"test.swift").write " class aClass {}  ) ;"
    assert_match(  "@startuml" \
                  "' STYLE START" \
                  "hide empty members" \
                  "skinparam shadowing false" \
                  "' STYLE END" \
                  "" \
                  "" \
                  "class \"aClass\" as aClass << (C, DarkSeaGreen) >>" \
                  "}" \
                  "" \
                  "" \
                  "@enduml",
                  shell_output("#{bin}/swiftplantuml test.swift --output consoleOnly").chomp
    )
  end
```

This should work but when I run `brew test swiftplantuml` I receive the following error on my local machine:

```bash
==> Testing swiftplantuml
==> /usr/local/Cellar/swiftplantuml/0.5.0/bin/swiftplantuml test.swift --output consoleOnly
Error: swiftplantuml: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected: 0
  Actual: nil
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/minitest-5.14.4/lib/minitest/assertions.rb:183:in `assert'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/minitest-5.14.4/lib/minitest/assertions.rb:218:in `assert_equal'
/usr/local/Homebrew/Library/Homebrew/formula_assertions.rb:26:in `shell_output'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/swiftplantuml.rb:28:in `block in <class:Swiftplantuml>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2097:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:550:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2096:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:949:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2095:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2348:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2089:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

Testing manually, ie. creating test.swft and run `/usr/local/Cellar/swiftplantuml/0.5.0/bin/swiftplantuml test.swift --output consoleOnly` in terminal then this works as expected.

Is it my local system or do you see an easy-to-spot beginner mistake? Sorry, but I am not very familiar with Ruby

Reply:
Try using `EOS` like other formulae do when they need to define a multiline string.

Reply:
I found the issue. Similar to another command line tool, `SwiftLint`, I am using a [dependency](https://github.com/jpsim/SourceKitten) that cannot work in a sandboxed environment like Homebrew CI. 

To solve this issue it is possible to add an environment variable and instrument the command-line tool to disable/notUse that dependency.

https://github.com/Homebrew/homebrew-core/blob/b4a4beeca8d3b4c30906b1016bd344961403978c/Formula/swiftlint.rb#L30

However, for my command-line tool this dependency is essential for regular functionality. I'd like to ask you to consider the version test as being sufficient. Another tool, `sourcery`, apparently was successful with such appeal.

https://github.com/Homebrew/homebrew-core/blob/b887aa84fd9f8041aa9cefe87d053e48ea53e59d/Formula/sourcery.rb#L27
https://github.com/Homebrew/homebrew-core/blob/b887aa84fd9f8041aa9cefe87d053e48ea53e59d/Formula/sourcery.rb#L28

Reply:
What in the sandbox is breaking the tool?

Reply:
I think sourcekit needs its own sandbox and won't run without one. And you can't run a sandbox-within-a-sandbox.

Reply:
@carlocab is correct, SourceKit requires not being sandboxed. SourceKit, the ability of interpreting source code, is essential for my tool `swiftplantuml`. Same situation as for `sourcery` which got [approved](https://github.com/Homebrew/homebrew-core/pull/69794) eventually to having "just" a version test.

Reply:
@chenrui333 @SMillerDev is it ok publish the formula with a version check only in this particular case?

If not then I could introduce an environment variable and instrument my command-line tool to disable/notUse the dependency but this would probably skip up to 70% of the code.

Reply:
Sounds good to me.